nodes/src/nodes/tool_filesystem/IInstance.py

```
@@ -0,0 +1,346 @@
+        return bool(getattr(self.IGlobal, flag, False))
+
+    def _check_ready(self) -> None:
+        """Verify the FileStore was successfully initialised in beginGlobal()."""
+        if self.IGlobal.file_store is None:
+            raise ValueError('filesystem tool is not available: ROCKETRIDE_CLIENT_ID is missing or the account store failed to initialise (check pipeline logs)')
+
+    def _check_path(self, path: str) -> None:
+        """Enforce the configured path whitelist (if any)."""
+        patterns = self.IGlobal.path_patterns or []
+        if patterns and not any(p.search(path) for p in patterns):
+            raise ValueError(f'path {path!r} does not match any allowed path pattern')
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)